service/resource/configmap/resource_test.go

```diff
@@ -991,19 +985,7 @@ func Test_Resource_CloudConfig_GetUpdateState(t *testing.T) {
 	}
 
 	for i, tc := range testCases {
-		ctx := context.Background()
-
-		ctx = updateallowedcontext.NewContext(ctx, make(chan struct{}))
-		ctx = updatenecessarycontext.NewContext(ctx, make(chan struct{}))
-
-		if tc.SetCtxUpdateAllowed {
-			updateallowedcontext.SetUpdateAllowed(ctx)
-		}
-		if tc.SetCtxUpdateNecessary {
-			updatenecessarycontext.SetUpdateNecessary(ctx)
-		}
-
-		createState, deleteState, updateState, err := newResource.GetUpdateState(ctx, tc.Obj, tc.CurrentState, tc.DesiredState)
+		createState, deleteState, updateState, err := newResource.GetUpdateState(tc.Ctx, tc.Obj, tc.CurrentState, tc.DesiredState)
```

Comment:
Same here, IMO subtests can help to determine easily what failed in case of error, also to give a descriptive name to each case.

Reply:
I can check on this.